### PR TITLE
Multiserver

### PR DESCRIPTION
**Pull request by Finanzcreeper**
Fixed Leaks
Added multiserver functionality

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1,26 +1,59 @@
 #include "server/Server.hpp"
 #include <vector>
 #include "parsers/ConfigParse.hpp"
+#include <signal.h>
+
+bool endserver = false;
+
+void WebKill(int signal) {
+	if (signal == 2) {
```
Comment (hoeftint) [inline main.cpp]: Using `SIGINT` instead of 2 would be more readable

File: server/Server.cpp [hunk 2]
```diff
@@ -159,5 +165,6 @@ void Server::checkConnectionsForTimeout() {
 }
 
 Server::~Server() {
+	std::cout << "IM DEAD" << std::endl;
```
Comment (hoeftint) [inline server/Server.cpp]: Not sure if this is a remainer from debugging, but when we print a message in the deconstructor, would be nice to get information which server is deconstructed, for example by printing the ip adress

File: server/Server.hpp [hunk 3]
```diff
@@ -28,10 +30,20 @@
 class Server {
 private:
 
+	std::vector<pollfd>::iterator it;
+	std::map<int, connection>::iterator mt;
+	std::map<int, Response>::iterator resps;
```
Comment (hoeftint) [inline server/Server.hpp:35]: we should also rename `Response` to `response` maybe to have matching cases

**Review by hoeftint — APPROVED** [review]
Great job with the  new loops, just some minor comments for style

File: main.cpp (hunk 1 above)
Comment (Finanzcreeper) [reply main.cpp]: yes, done

File: server/Server.cpp (hunk 2 above)
Comment (Finanzcreeper) [reply server/Server.cpp]: was for debugging, i removed it for now.

File: server/Server.hpp (hunk 3 above)
Comment (Finanzcreeper) [reply server/Server.hpp:35]: we can refactor that later.

**Pull request merged**